src/containers/Tasks.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { addTask, deleteTask, completeTask, editState, editTask } from '../actions/index';
+import '../css/tasks.css';
+
+
+class Tasks extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      editing: false
+    };
+    this.handleEditing = this.handleEditing.bind(this);
+  }
+
+  handleEditing = () => {
+    this.setState({editing: !this.state.editing});
+  }
+
+  edit = (e) => {
+    const str = this.refs.editTask.value;
+    if (/\S/.test(str)) {
+      this.props.editTask(str, e);
+    }
+  }
+
+ add = () => {
+   const str = this.refs.addTask.value;
+   if (/\S/.test(str)) {
+     this.props.addTask(str, this.props.user.id);
+   }
+   this.refs.addTask.value = '';
+ }
+
+  showTasks() {
+    return this.props.tasks.map((task, i) => {
+      if (!task.edit) {
+        return(
+          <li key={i} title='Done?' className='task__wrapper'>
```

Comment:
don't use index as a key, have a read https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318